src/electionguard/elgamal.cpp

```diff
@@ -514,96 +521,108 @@ namespace electionguard
 
 #pragma endregion // HashedElGamalCiphertext
 
-    unique_ptr<HashedElGamalCiphertext>
-    hashedElgamalEncrypt(std::vector<uint8_t> message, const ElementModQ &nonce,
-                         const ElementModP &publicKey, const ElementModQ &encryption_seed,
-                         padded_data_size_t max_len, bool allow_truncation,
-                         bool shouldUsePrecomputedValues /* = false */)
+    vector<uint8_t> formatMessage(vector<uint8_t> message,
+                                  HASHED_CIPHERTEXT_PADDED_DATA_SIZE max_len, bool allow_truncation)
     {
-        vector<uint8_t> ciphertext;
-        vector<uint8_t> plaintext_on_boundary;
+        if (max_len == 0 || max_len > HASHED_CIPHERTEXT_PADDED_DATA_SIZE::BYTES_512) {
+            throw invalid_argument("HashedElGamalCiphertext::encrypt max_len is invalid");
+        }
+
+        // TODO: HACK: ISSUE #358: we need to check the modulo of the max_len matches the block length
+        // and handle the indicator size truncation inline inside this function
 
         // padding scheme is to concatenate [length of the padding][plaintext][padding bytes of 0x00]
         // padding bytes 0x00 are padded out to the first HASHED_CIPHERTEXT_BLOCK_LENGTH boundary
         // past max_len. So if max_len is 62 then it will pad to the 64 byte boundary
-        if (max_len != NO_PADDING) {
-            uint16_t pad_len = 0;
-            uint16_t pad_len_be = 0;
-
-            if (allow_truncation && (message.size() > max_len)) {
-                // truncate the data
-                // insert length in big endian form
-                plaintext_on_boundary.insert(plaintext_on_boundary.end(), (uint8_t *)&pad_len_be,
-                                             (uint8_t *)&pad_len_be + sizeof(pad_len_be));
-                // insert plaintext
-                plaintext_on_boundary.insert(plaintext_on_boundary.end(), &message.front(),
-                                             &message.front() + max_len);
-            } else {
-                if (message.size() > max_len) {
-                    throw invalid_argument(
-                      "HashedElGamalCiphertext::encrypt the plaintext is greater than max_len");
-                }
-
-                uint16_t pad_len = max_len - message.size();
-                uint16_t pad_len_be = htobe16(pad_len);
 
-                std::vector<uint8_t> padding(pad_len, 0);
+        vector<uint8_t> formattedMessage;
```

Comment:
none of the actual behavior in here changed, i just refactored the method into 3 different methods so it allowed removing NO_PADDING.